Common/Retranslator/src/Retranslator.cpp

```diff
@@ -0,0 +1,210 @@
+#include "Retranslator.hpp"
+#include <typeinfo>
+#include <iostream>
+using namespace AST;
+
+std::unique_ptr<Retranslator> Retranslator::instance = nullptr;
+
+ASTTree Retranslator::parseTree(std::vector<Shape*> const& vec) const{
+    
+    Node* root = new Node("root");
+    root->addChild(new Node("STARTGRAPH"));
+
+    for (const auto& elem : vec)
+    {
+        Node* cur = new Node("STATEMENT");
+        root->addChild(cur);
+        // if (dynamic_cast<Note*>(elem.second))
+        // {
+        //     root->addChild(makeNote(elem));
+        // }
+        if (const Line* const line = dynamic_cast<const Line* const>(elem))
+        {
+            cur->addChild(makeLink(line));
+        }
+        else if (const Graph* const graph = dynamic_cast<const Graph* const>(elem))
+        {
+            cur->addChild(makeGraph(graph));
+        }
+        else 
+        {
+            cur->addChild(makeObject(elem));
+        }
+    }
+
+    root->addChild(new Node("ENDGRAPH"));
+
+    return ASTTree(root);
+}
+
+Node* makeNodeWithProperty(std::string key, std::string value, bool isText){
+    Node* property = new Node("property");
+    property->addChild(new Node("PROPERTY_KEY"));
+    property->childNodes[0]->addChild(new Node(key));
+    property->addChild(new Node("="));
+    if (isText)
+        property->addChild(new Node("TEXT"));
+    else 
+        property->addChild(new Node("NUMBER"));
+    property->childNodes[2]->addChild(new Node(value));
+
+    return property;
+}
+
+void addProperty(Node* node, std::string const& key, std::string const& value, bool isText = false)
+{
+    node->addChild(makeNodeWithProperty(key, value, isText));
+    node->addChild(new Node(";"));
+}
+
+void addBasicParams(Node* node, const Shape* const shape)
+{
+    addProperty(node, "text", shape->text, true);
+    addProperty(node, "сolor", std::to_string(static_cast<short>(shape->style.color)));
```

Comment:
```suggestion
    addProperty(node, "сolor", std::to_string(static_cast<short>(shape->style.color)), true);
```
Кажется, что это текст